Converts a compile time type check that generates an unused function
warning to an unused variable warning which is suppressed when
building the kernel.

This patch allows this code to be compiled with either gcc or clang.

Author: Mark Charlebois <charlebm@gmail.com>
Signed-off-by: Mark Charlebois <charlebm@gmail.com>
---

diff --git a/include/linux/moduleparam.h b/include/linux/moduleparam.h
index ea36486..cf65778 100644
--- a/include/linux/moduleparam.h
+++ b/include/linux/moduleparam.h
@@ -114,10 +114,13 @@ struct kparam_array
  * same, but that's harder if the variable must be non-static or is inside a
  * structure.  This allows exposure under a different name.
  */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-function"
 #define module_param_named(name, value, type, perm)			   \
 	param_check_##type(name, &(value));				   \
 	module_param_cb(name, &param_ops_##type, &value, perm);		   \
 	__MODULE_PARM_TYPE(name, #type)
+#pragma clang diagnostic pop
 
 /**
  * module_param_cb - general callback for a module/cmdline parameter
